k8s-pb-codegen/src/main.rs

```diff
@@ -50,10 +60,41 @@ fn main() -> Result<()> {
                     }
                 }
             }
+
+            let mut parts = pkg.split('.').collect::<Vec<_>>();
+            while !parts.is_empty() {
+                let module = parts.pop().unwrap();
+                let parent = parts.join("/");
+                module_tree
+                    .entry(parent)
+                    .or_insert_with(HashSet::new)
+                    .insert(module.to_owned());
+            }
+            pkgs.push(pkg);
         }
     }
 
-    // Generate code in `src/` by reading files in `OUT_DIR`?
-    // Need to create `mod.rs` file for each level based on the filename, and write generated code in correct file.
+    for (k, v) in module_tree {
+        let dst = if k.is_empty() {
+            target_dir.join("lib.rs")
+        } else {
+            target_dir.join(format!("{}/mod.rs", k))
+        };
+        std::fs::create_dir_all(dst.parent().unwrap())?;
+        let mut lines = v
+            .into_iter()
+            .map(|m| format!("pub mod {};", m))
+            .collect::<Vec<_>>();
+        lines.sort();
```

Comment:
might be able to avoid this `sort()` if we use `BTreeSet` rather than `HashSet` in the map